lib/active_record/dynamic_timeout/extensions/base_extension.rb

```diff
@@ -8,15 +8,17 @@ module BaseExtension
     extend ActiveSupport::Concern
 
     module ClassMethods
-      def with_timeout(timeout)
-        (timeout.is_a?(Integer) || timeout.nil?) or raise ArgumentError, "timeout must be an Integer or NilClass, got: `#{timeout.inspect}`"
-        timeout_stack << timeout
+      # @param [Numeric, NilClass] timeout_seconds The timeout in seconds, or nil to disable the timeout.
+      def with_timeout(timeout_seconds)
+        (timeout_seconds.is_a?(Numeric) || timeout_seconds.nil?) or raise ArgumentError, "timeout_seconds must be Numeric or NilClass, got: `#{timeout_seconds.inspect}`"
```

Comment:
I think this `raise` will land in the `ensure` below. (This is a common bug with `ensure` so I always check for it.)

I believe the cleanest fix is to check the precondition first, then have an explicit `begin`/`ensure`/`end` block.
```ruby
begin
  timeout_stack << timeout_seconds
  yield
ensure
  timeout_stack.pop
end

Reply:
Rather than "must be Numeric or NilClass", how about "must be Numeric or nil"? I think that would be a bit more natural in Ruby `NilClass` is 1-to-1 with `nil`.

Reply:
Nice catch, fixed: https://github.com/Invoca/activerecord-dynamic_timeout/pull/1/commits/22760e04130e7103e1e32fa1f458f66f26c99aa4